src/Symfony/Component/DependencyInjection/Loader/FileLoader.php

```diff
@@ -56,10 +56,27 @@ public function registerClasses(Definition $prototype, $namespace, $resource, $e
 
         $classes = $this->findClasses($namespace, $resource, $exclude);
         // prepare for deep cloning
-        $prototype = serialize($prototype);
+        $serializedPrototype = serialize($prototype);
+        $interfaces = array();
+        $interfaceImplementations = array();
 
         foreach ($classes as $class) {
-            $this->setDefinition($class, unserialize($prototype));
+            if (interface_exists($class, false)) {
+                $interfaces[] = $class;
+            } else {
+                $this->setDefinition($class, unserialize($serializedPrototype));
+                foreach (class_implements($class, false) as $interface) {
+                    $interfaceImplementations[$interface][] = $class;
+                }
+            }
+        }
+        foreach ($interfaces as $interface) {
+            if (!isset($interfaceImplementations[$interface]) || isset($interfaceImplementations[$interface][1])) {
+                continue;
+            }
+            $this->container->setAlias($interface, $interfaceImplementations[$interface][0])
+                ->setPublic($prototype->isPublic())
```

Comment:
As these aliases are meant to be used for autowiring, what about always creating them as private ?

Reply:
OK for me, updated